editor/editor_inspector.cpp

```diff
@@ -2297,6 +2297,9 @@ void EditorInspector::_edit_set(const String &p_name, const Variant &p_value, bo
 					undo_redo->add_undo_method(r, "setup_local_to_scene");
 				}
 			}
+
+			undo_redo->add_do_method(r, "emit_changed");
+			undo_redo->add_undo_method(r, "emit_changed");
```

Comment:
This is required for the label to update when a value is changed in the inspector.

I can submit this as a separate PR if needed... If a resource is changed in the inspector, `emit_changed` should be called, right? Otherwise there is essentially no good way for my plugin to know when the resource was changed.

Reply:
Emit changed is used in the resource itself as a common way to notify yourself that something changed and its implemented only if you need it. If this is the case, it sounds like you should just add this on InputEvent instead.

Reply:
Wouldn't that require adding `emit_changed` to every single method though, on every input event? `set_keycode`, `set_physical_keycode`, `set_alt_pressed`, `set_shift_pressed`, `set_button_index (for the mouse)` etc. That is a lot of added code - is this the way it should be?